test/mocks.ts

```diff
@@ -0,0 +1,35 @@
+jest.mock('@actions/core', () => coreMock);
+import { Options } from "../src/main";
+
+export const coreMock = {
+  setFailed: jest.fn(),
+  setOutput: jest.fn(),
+  getBooleanInput: jest.fn(),
+  getInput: jest.fn(),
+  error: jest.fn(),
+  warning: jest.fn(),
+  info: jest.fn(),
+  debug: jest.fn(),
+};
+
+export function mockInputs(options: Options) {
+  coreMock.getInput.mockImplementation(name => {
+    if (name === 'scope') return options.scope;
+    if (name === 'subscriptionId') return options.subscriptionId;
+    if (name === 'region') return options.region;
+    if (name === 'resourceGroupName') return options.resourceGroupName;
+    if (name === 'template') return options.template;
+    if (name === 'deploymentMode') return options.deploymentMode;
+    if (name === 'deploymentName') return options.deploymentName;
+    if (name === 'parameters') return options.parameters;
+    if (name === 'managementGroupId') return options.managementGroupId;
+    if (name === 'additionalArguments') return options.additionalArguments;
+
+    throw new Error(`Unexpected input: ${name}`);
```

Comment:
Can simplify the code block to
```suggestion
    const option = options[name as keyof Options];

    if (option === undefined) {
      throw new Error(`Unexpected input: ${name}`);
    }

    return option;
```